snowfall/training/ctc_graph.py

```diff
@@ -85,9 +84,10 @@ def compile_one_and_cache(self, text: str) -> k2.Fsa:
         tokens = (token if token in self.words else self.oov
                   for token in text.split(' '))
         word_ids = [self.words[token] for token in tokens]
-        fsa = k2.linear_fsa(word_ids)
-        decoding_graph = k2.connect(k2.intersect(fsa, self.L_inv)).invert_()
+        label_graph = k2.linear_fsa(word_ids)
+        decoding_graph = k2.connect(k2.intersect(label_graph,
+                                                 self.L_inv)).invert_()
```

Comment:
Actually, instead of adding the disambig symbols into the ctc_topo, we can handle them here.
  (i) we can use compose instead of intersection and double inversion, here.
  (ii) after intersecting, we can replace all the disambig symbols with epsilon and then remove epsilons, and then connect.

Reply:
Oh, wait... now I see that train.py is loading L.fst not L_disambig.fst.  So there may be no disambig symbols to deal with.